python-service/src/health_aggregator.py

```
@@ -0,0 +1,97 @@
+        )
+        try:
+            urllib.request.urlopen(req, timeout=5)
+            return True
+        except urllib.error.URLError:
+            return False
+
+    def get_history(self, service: str, limit: int = 100) -> List[Dict]:
+        conn = sqlite3.connect(self.db_path)
+        rows = conn.execute(
+            f"SELECT data, ts FROM health_snapshots WHERE service = '{service}' "
+            f"ORDER BY ts DESC LIMIT {limit}"
```

Ruff (0.15.15)

[error] Possible SQL injection vector through string-based query construction

(S608)